src/tally.F90

```diff
@@ -395,6 +402,195 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
+            nuc => nuclides(i_nuclide)
+
+            ! Check if the delayed group filter is present
+            if (dg_filter > 0) then
+
+              ! Loop over all delayed group bins and tally to them individually
+              do d_bin = 1, t % filters(dg_filter) % n_bins
+
+                ! Get the delayed group for this bin
+                d = t % filters(dg_filter) % int_bins(d_bin)
+
+                ! Compute the yield for this delayed group
+                yield = yield_delayed(nuc, p % E, d)
+
+                ! Compute the score and tally to bin
+                score = micro_xs(i_nuclide) % fission * yield &
+                     * nu_delayed(nuc, p % E) * atom_density * flux
+                call score_fission_delayed_dg(t, d_bin, score, score_index)
+              end do
+              cycle SCORE_LOOP
+            else
+
+              ! If the delayed group filter is not present, compute the score
+              ! by multiplying the delayed-nu-fission macro xs by the flux
+              score = micro_xs(i_nuclide) % fission * nu_delayed(nuc, p % E)&
+                   * atom_density * flux
+            end if
+
+          ! Tally is on total nuclides
+          else
+
+            ! Get pointer to current material
+            mat => materials(p % material)
+
+            ! Check if the delayed group filter is present
+            if (dg_filter > 0) then
+
+              ! Loop over all nuclides in the current material
+              do l = 1, mat % n_nuclides
+
+                ! Get atom density
+                atom_density_ = mat % atom_density(l)
+
+                ! Get index in nuclides array
+                i_nuc = mat % nuclide(l)
+
+                ! Loop over all delayed group bins and tally to them individually
+                do d_bin = 1, t % filters(dg_filter) % n_bins
+
+                  ! Get the delayed group for this bin
+                  d = t % filters(dg_filter) % int_bins(d_bin)
+
+                  ! Get the current nuclide
+                  nuc => nuclides(i_nuc)
+
+                  ! Get the yield for the desired nuclide and delayed group
+                  yield = yield_delayed(nuc, p % E, d)
+
+                  ! Compute the score and tally to bin
+                  score = micro_xs(i_nuc) % fission * yield &
+                       * nu_delayed(nuc, p % E) * atom_density_ * flux
+                  call score_fission_delayed_dg(t, d_bin, score, score_index)
+                end do
+              end do
+              cycle SCORE_LOOP
+            else
+
+              score = ZERO
+
+              ! Loop over all nuclides in the current material
+              do l = 1, mat % n_nuclides
+
+                ! Get atom density
+                atom_density_ = mat % atom_density(l)
+
+                ! Get index in nuclides array
+                i_nuc = mat % nuclide(l)
+
+                ! Get the current nuclide
+                nuc => nuclides(i_nuc)
```

Comment:
I'm getting the following compiler error when I make that switch:

    Error: Actual argument for 'nuc' must be a pointer at (1)

Is there a way to pass the pointer to the nuclide with the change you recommended?